velox/buffer/Buffer.h

```diff
@@ -172,6 +175,32 @@ class Buffer {
     return os;
   }
 
+  /// Slice a buffer with specific type T.
+  /// For boolean type and if the 'offset' is not multiple of 8, return a
+  /// shifted copy, new buffer is allocated from 'pool'.
+  /// Otherwise return a BufferView into the original buffer (with shared
+  /// ownership of original buffer).
+  ///
+  /// @param buf A pointer to the buffer to be sliced. If this is null, the
+  /// function returns a null pointer.
+  /// @param offset The starting point in the buffer from which the slice
+  /// begins.
+  /// @param length The number of elements to include in the slice.
+  /// @param pool A pointer to a memory pool for allocating new buffers,
+  /// required if a new buffer needs to be created.
+  template <typename T>
+  static BufferPtr slice(
+      const BufferPtr& buf,
+      size_t offset,
+      size_t length,
+      memory::MemoryPool* pool) {
+    if (!buf) {
```

Comment:
Perhaps, require the caller to pass non-null buffer.

Reply:
There is test case which uses the null buffer, `LazyVectorTest.lazySlice`.

Reply:
This is a new API, so no existing test can call it. The callers of the existing API may need to be updated to not pass null.

Reply:
I think in this PR I add new API Buffer::slice to replace the BaseVector::sliceBuffer. And the old API is mostly used in vector::slice (BaseVector, DictionaryVector, FlatVector, ComplexVector) to slice the indices buffer, nulls buffer or value buffer. I'm not sure indices buffer, but nulls buffer may be nullptr in some cases. So I think we need to maintain old behavior: return null if buffer is nullptr. Please let me know if I need to move this check to caller.
Thanks.

Reply:
It would be nice to move null check to the callers.